### PR TITLE
Use etag to cache youtube responses

### PR DESCRIPTION
**Pull request by dotted**
This should drastically reduce quota usage for youtube api.

Questions:
- ~Should the old cache logic be removed?~
- Should we dump the `this.etags` into the DB to persists the etags further reducing quota cost?

Todo:

- [x] Fix tests

### CODE REVIEW

**Comment by johnpyp** [comment]
Since the bot is generally long-running for 24h+ I doubt it's worth it to dump etags into the db, unless we're still running into other api issues after this change.

LGTM

File: lib/services/youtube.js [hunk 1]
```diff
@@ -50,20 +61,25 @@ class YouTube {
   }
 
   getChannelStatus() {
-    const now = moment().unix();
+    /* const now = moment().unix();
```
Comment (johnpyp) [inline lib/services/youtube.js:64]: Should probably be deleted

**Comment by shadedred** [comment]
This should hopefully provide better caching, sufficient enough to keep us under the quota. The next solution here, should this still hit the quota, is to not us the search API, and get destiny to generate an authenticated key on his account itself that can use the https://developers.google.com/youtube/v3/live/docs/liveBroadcasts/list list (my broadcasts) endpoint, eliminating the search call, which is about 160 times more expensive than every other call in terms of quota.

**Comment by dotted** [comment]
Pretty sure dumping `this.etags` into the DB and loading it on startup is much less involved than trying to get Destiny to get us that key. 🤣

**Comment by shadedred** [comment]
I agree 😆 but lets give this a shot and see how the quotas fair tomorrow

**Pull request merged**